File: ggts/parsers.py
import re
from typing import List, Optional

class TicketParser:
    JIRA_REGEX = re.compile(r'[A-Z]+-\d+')
    GITHUB_REGEX = re.compile(r'(?:^|\s)#(\d+)')

    @staticmethod
    def extract_tickets(message: str) -> List[str]:
        tickets = []
        
        # Extract Jira tickets
        jira_matches = TicketParser.JIRA_REGEX.findall(message)
        tickets.extend(jira_matches)
        
        # Extract GitHub issues
        github_matches = TicketParser.GITHUB_REGEX.findall(message)
        tickets.extend([f"#{m}" for m in github_matches])
        
        return list(set(tickets))

File: src/parsers.ts
export class TicketParser {
  // Jira: Project-Number (e.g. PROJ-123, IT-456). Require at least 2 chars for project.
  static JIRA_REGEX = /\b([A-Z]{2,}-\d+)\b/g;

  static extractTickets(message: string): string[] {
    const tickets = new Set<string>();

    let match;
    while ((match = this.JIRA_REGEX.exec(message)) !== null) {
      tickets.add(match[1]);
    }

    return Array.from(tickets);
  }
}
